README.md

```diff
@@ -334,6 +334,80 @@ class Tracking < Coach::Middleware
 end
 ```
 
+# Coach CLI (alpha)
+
+_Note: The `coach` CLI is currently considered **alpha** stability, though we welcome bug
+reports if you find it doesn't work with your application._
+
+As well as the library, the Coach gem now comes with a command line tool - `coach`.
+
+When working in a large codebase that uses Coach, one of the challenges you may run into
+is understanding the `provide`/`require` graph made up of all the middleware chains you've
+built. While the library enforces the correctness of those chains at boot time, it doesn't
+help you understand those dependencies. That's where the `coach` CLI comes in!
+
+Currently, the `coach` CLI supports two commands.
+
+## `find-provider`
+
+`find-provider` is the simpler of the two commands. Given the name of a Coach middleware
+and a value that it requires, it outputs the name of the middleware that provides it.
+
+```bash
+$ bundle exec coach find-provider HelloUser user
+Value `user` is provided to `HelloUser` by:
+
+Authentication
+```
+
+If there are multiple middlewares in the chain that provide the same value, all of them
+will be listed.
+
+## `find-chain`
+
+`find-chain` is the more advanced of the two commands, and is most useful in larger
+codebases. Given the name of a Coach middleware and a value it requires, it outputs the
+chains of middleware between the specified middleware and the one that provides the
+required value.
+
+```bash
+# Note that we've assumed an intermediate middleware - `UserDecorator` exists in this
+# example to make the functionality of the command clearer.
+$ bundle exec coach find-chain HelloUser user
+Value `user` is provided to `HelloUser` by:
+
+HelloUser -> UserDecorator -> Authentication
```

Comment:
Are the arrows pointing the right way in the output here? I would have thought either `Route <- Middleware B <- Middleware A` or maybe this could be expressed in a hierarchical format?

Reply:
I'm gonna reverse the list. Originally I was thinking about it in terms of dependencies (i.e. traversing up the chain), but when you're talking about providing a value it makes sense to talk about it in the other direction.